rdflib/plugins/sparql/processor.py

```diff
@@ -66,16 +57,116 @@ class SPARQLProcessor(Processor):
     def __init__(self, graph):
         self.graph = graph
 
+    def check(s):
+        pass
+
+    def Describe(self, strOrQuery):
+        pass
+
+    def clean_query(self, strOrQuery):
+
+        temp = strOrQuery.replace("SELECT *", "SELECT **SELECT *").replace("\n", "")
```

Comment:
This seems to be based on string manipulation which is not ideal and I don't think strOrQuery will always be a sting. It should be done on the AST instead, and I'm also not sure I understand the reasoning here, we should not clean the query to handle nested subqueries, we should just handle them.